src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -138,20 +138,42 @@ func main() {
 		wg.Add(1)
 	}
 	invoker := common.GetInvoker(kc, logger)
-	cluster := assistedController.SetReadyState()
-	if cluster == nil && invoker == common.InvokerAgent {
-		// When the agent-based installer installs a SNO cluster, assisted-service
-		// will never be reachable because it will not be running after the boostrap
-		// node reboots to become the SNO cluster. SetReadyState will never be
-		// able to get a connection to assisted-service in this circumstance.
-		//
-		// Instead of exiting with panic with "invalid memory address or nil pointer
-		// dereference" and having the controller restart because cluster is nil,
-		// log warning and exit 0. Otherwise the controller will keep restarting
-		// leaving the controller in a running state, even through the cluster
-		// has finished install.
-		logger.Warnf("SetReadyState timed out fetching cluster from assisted-service, invoker = %v", invoker)
-		return
+	var cluster *models.Cluster
+	var platformType string
+	if invoker == common.InvokerAgent {
+		// With the agent-based installer, assisted-service runs on the bootstrap node.
+		// We do not need to wait for an extended amount of time to reach assisted-service
+		// because it should be running on the local network.
+		cluster = assistedController.SetReadyState(5 * time.Minute)
+
+		if cluster == nil && Options.ControllerConfig.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
```

Comment:
Will not you always be enter this part in sno mode? Why we actually should even try to get cluster in this case?

Reply:
You're right, the SNO case doesn't need cluster.